[dynamo][hf_bigbird] Actually graph break on tensor.unsqueeze_/resize_

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100071
* __->__ #99986


Currently, we return `unimplemented` w/o a graph break on seeing a x.unsqueeze_ when x is input. This essentially means we fall back to the original frame.

This PR actually graph breaks so that we can generate the continuation frame for the rest of the function. Instead of graph breaking at LOAD_ATTR, we delay the graph break to the actual CALL_FUNCTION, where its cleaner to graph break.

cc @soumith @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

anijain2305 committed:
[dynamo] graph break on load_attr (tensor.unsqueeze_)

[ghstack-poisoned]

jansel commented on torch/_dynamo/symbolic_convert.py:
```diff
@@ -1133,6 +1133,7 @@ def CALL_METHOD(self, inst):
         fn = self.pop()
         self.call_function(fn, args, {})
 
+    @break_graph_if_unsupported(push=0)
```
`push=0` doesn't sound right

anijain2305 committed:
Update on "[dynamo] graph break on load_attr (tensor.unsqueeze_)"




cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Actually graph break on tensor.unsqueeze_/resize_"



Currently, we return `unimplemented` w/o a graph break on seeing a x.unsqueeze_ when x is input. This essentially means we fall back to the original frame.

This PR actually graph breaks so that we can generate the continuation frame for the rest of the function. Instead of graph breaking at LOAD_ATTR, we delay the graph break to the actual CALL_FUNCTION, where its cleaner to graph break.

cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Actually graph break on tensor.unsqueeze_/resize_"



Currently, we return `unimplemented` w/o a graph break on seeing a x.unsqueeze_ when x is input. This essentially means we fall back to the original frame.

This PR actually graph breaks so that we can generate the continuation frame for the rest of the function. Instead of graph breaking at LOAD_ATTR, we delay the graph break to the actual CALL_FUNCTION, where its cleaner to graph break.

cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

anijain2305 commented:
@jansel Made some changes. Can you PTAL?

anijain2305 commented:
@pytorchbot merge

eellison commented on torch/_dynamo/variables/tensor.py:
```diff
@@ -182,7 +182,8 @@ def var_getattr(self, tx, name):
         # It's hard to get resize_() on graph input work properly across
         # dynamo/aot/inductor, just fall back.
         if name in ("resize_", "unsqueeze_") and self.source is not None:
```
can you use the [inplace tag](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/tags.yaml#L3-L5) list ? `resize_` and `unsqueeze` is an incomplete list

jansel replied:
This list should only be for metadata mutation, not all mutation.

eellison replied:
Yea, sorry I linked to `inplace_vew`, which is metadata only. mistyped
